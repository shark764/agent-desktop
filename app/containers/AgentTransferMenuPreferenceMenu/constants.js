/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentTranferMenu constants
 *
 */

// Redux constants
export const SET_AGENTS_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/SET_AGENTS_TRANSFER_MENU_PREFERENCE';

//  Redux Queues Constants
export const SET_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/SET_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE';

//  Redux Transfer Lists Constants
export const SET_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/SET_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE';

// Saga Constants
export const INITIALIZE_TRANSFER_MENU_PREFERENCES =
  'app/AgentTransferMenuPreferenceMenu/INITIALIZE_TRANSFER_MENU_PREFERENCES';
export const TOGGLE_AGENTS_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_AGENTS_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE';
export const TOGGLE_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE =
  'app/AgentTransferMenuPreferenceMenu/TOGGLE_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE';
