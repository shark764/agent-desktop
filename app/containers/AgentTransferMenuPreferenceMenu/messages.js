/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 * AgentNotificationsMenu Messages
 *
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  agents: {
    id: 'app.containers.AgentTransferMenuPreferenceMenu.agents',
    defaultMessage: 'Agents',
  },
  queues: {
    id: 'app.containers.AgentTransferMenuPreferenceMenu.queues',
    defaultMessage: 'Queues',
  },
  transferLists: {
    id: 'app.containers.AgentTransferMenuPreferenceMenu.transferLists',
    defaultMessage: 'Transfer Lists',
  },
});
