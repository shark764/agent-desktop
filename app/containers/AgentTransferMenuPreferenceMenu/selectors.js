/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */
import { createSelector } from 'reselect';
import { selectQueues } from 'containers/AgentDesktop/selectors';
import { selectUserAssignedTransferLists, selectUserAssigNonVoiceTransLists } from 'containers/TransferMenu/selectors';

export const selectTransferMenuPreferences = (state) =>
  state.get('transferMenuPreferences');

export const selectSelectedQueues = createSelector(
  selectTransferMenuPreferences,
  (transferMenuPreference) =>
    transferMenuPreference.get('selectedQueues').toJS()
);

export const selectSelectedTransferLists = createSelector(
  selectTransferMenuPreferences,
  (transferMenuPreference) =>
    transferMenuPreference.get('selectedTransferLists').toJS()
);

export const selectAgentsPreferences = (state) =>
  selectTransferMenuPreferences(state).get('agentsTransferMenu');

export const selectQueuesPreferences = createSelector(
  selectQueues,
  selectSelectedQueues,
  (userQueues, selectedQueues) => userQueues.length === selectedQueues.length
);

export const selectVisibleQueues = createSelector(
  selectQueues,
  selectSelectedQueues,
  (userQueues, selectedQueues) =>
    userQueues.filter((queue) => selectedQueues.includes(queue.id))
);

export const selectVisibleTransferLists = createSelector(
  selectUserAssignedTransferLists,
  selectSelectedTransferLists,
  (userTransferLists, selectedTransferLists) => {
    if (userTransferLists !== null) {
      return userTransferLists.filter((transferList) =>
        selectedTransferLists.includes(transferList.id)
      );
    } else {
      return [];
    }
  }
);

export const selectVisibleNonVoiceTransferLists = createSelector(
  selectUserAssigNonVoiceTransLists,
  selectSelectedTransferLists,
  (userTransferLists, selectedTransferLists) => {
    if (userTransferLists !== null) {
      return userTransferLists.filter((transferList) =>
        selectedTransferLists.includes(transferList.id)
      );
    } else {
      return [];
    }
  }
);

export const selectVisibleVoiceTransferLists = createSelector(
  selectUserAssignedTransferLists,
  selectSelectedTransferLists,
  (userTransferLists, selectedTransferLists) => {
    if (userTransferLists !== null) {
      return userTransferLists.filter((transferList) =>
        selectedTransferLists.includes(transferList.id)
      );
    } else {
      return [];
    }
  }
);

export const selectShowQueues = (state) =>
  selectTransferMenuPreferences(state).get('showQueues');

export const selectShowTransferLists = (state) =>
  selectTransferMenuPreferences(state).get('showTransferLists');

export const selectPreferenceMenuQueuesLoading = (state) =>
  selectTransferMenuPreferences(state).get('preferenceMenuQueuesLoading');

export const selectPreferenceMenuTransferListsLoading = (state) =>
  selectTransferMenuPreferences(state).get(
    'preferenceMenuTransferListsLoading'
  );
