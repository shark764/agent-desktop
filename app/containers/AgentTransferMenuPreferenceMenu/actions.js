/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentTransferMenuPreferenceMenu actions
 *
 */

import * as ACTIONS from './constants';

// Reducer actions
export function setAgentsTransferMenuPreference(agentsTransferMenu) {
  return {
    type: ACTIONS.SET_AGENTS_TRANSFER_MENU_PREFERENCE,
    agentsTransferMenu,
  };
}

//  Queues Reducer Actions
export function setShowQueuesTransferMenuPreference(showQueues) {
  return {
    type: ACTIONS.SET_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE,
    showQueues,
  };
}

export function toggleSelectedQueueTransferMenuPreferenceOnState(queue) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE_ON_STATE,
    queue,
  };
}

export function toggleAllSelectedQueuesTransferMenuPreferenceOnState(queues) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE_ON_STATE,
    queues,
  };
}

export function toggleSelectedQueues(queues) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE,
    queues,
  };
}

//  Transfer Lists Reducer Actions
export function setShowTransferListTransferMenuPreference(showTransferLists) {
  return {
    type: ACTIONS.SET_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    showTransferLists,
  };
}

export function toggleSelectedTransferListTransferMenuPreferenceOnState(
  transferList
) {
  return {
    type:
      ACTIONS.TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE_ON_STATE,
    transferList,
  };
}

export function toggleAllSelectedTransferListsTransferMenuPreferenceOnState(
  transferLists
) {
  return {
    type:
      ACTIONS.TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE_ON_STATE,
    transferLists,
  };
}

export function toggleSelectedTransferLists(transferLists) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    transferLists,
  };
}

// Saga actions
export function initializeTransferMenuPreferences() {
  return {
    type: ACTIONS.INITIALIZE_TRANSFER_MENU_PREFERENCES,
  };
}

export function toggleAgentsTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_AGENTS_TRANSFER_MENU_PREFERENCE,
  };
}

export function toggleShowQueuesTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE,
  };
}

export function toggleShowTransferListsTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
  };
}

export function setPreferenceMenuQueuesLoading(preferenceMenuQueuesLoading) {
  return {
    type: ACTIONS.SET_PREFERENCE_MENU_QUEUES_LOADING,
    preferenceMenuQueuesLoading,
  };
}

export function setPreferenceMenuTransferListsLoading(
  preferenceMenuTransferListsLoading
) {
  return {
    type: ACTIONS.SET_PREFERENCE_MENU_TRANSFER_LISTS_LOADING,
    preferenceMenuTransferListsLoading,
  };
}

export function toggleSelectedQueueTransferMenuPreference(queue) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE,
    queue,
  };
}

export function toggleAllSelectedQueuesTransferMenuPreference(queues) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE,
    queues,
  };
}

export function toggleSelectedTransferListTransferMenuPreference(transferList) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE,
    transferList,
  };
}

export function toggleAllSelectedTransferListsTransferMenuPreference(
  transferLists
) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    transferLists,
  };
}
