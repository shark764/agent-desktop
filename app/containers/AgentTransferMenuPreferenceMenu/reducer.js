/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentTransferMenuPreferenceMenu reducer
 *
 */

import { fromJS, List } from 'immutable';
import * as ACTIONS from './constants';

const initialState = fromJS({
  agentsTransferMenu: true,
  showQueues: false,
  showTransferLists: false,
  selectedQueues: [],
  selectedTransferLists: [],
  preferenceMenuQueuesLoading: false,
  preferenceMenuTransferListsLoading: false,
});

export default function transferMenuPreferencesReducer(
  state = initialState,
  action
) {
  switch (action.type) {
    //  Preferences Actions
    case ACTIONS.SET_AGENTS_TRANSFER_MENU_PREFERENCE:
      return state.set('agentsTransferMenu', action.agentsTransferMenu);

    //  Queues Actions
    //  Opens queues menu preference ui dropdown
    case ACTIONS.SET_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE:
      return state.set('showQueues', action.showQueues);
    //  Toggles one single queue item in dropdown
    case ACTIONS.TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE_ON_STATE:
      return state.update('selectedQueues', (queues) => {
        if (queues.includes(action.queue)) {
          return queues.filter((queue) => queue !== action.queue);
        } else {
          return queues.push(action.queue);
        }
      });
    //  Toggles all queues items in dropdown
    case ACTIONS.TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE_ON_STATE:
      return state.update('selectedQueues', (queues) => {
        if (action.queues.length > 0 && queues.size < action.queues.length) {
          const newQueues = action.queues.filter(
            (queue) => !queues.includes(queue)
          );
          return queues.push(...newQueues);
        } else {
          return List();
        }
      });
    //  Toggles all local storage queues items
    case ACTIONS.TOGGLE_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE:
      return state.set('selectedQueues', List(action.queues));
    //  Toggles the loading spinner for queues list in collapsibleMultiselect component
    case ACTIONS.SET_PREFERENCE_MENU_QUEUES_LOADING:
      return state.set(
        'preferenceMenuQueuesLoading',
        action.preferenceMenuQueuesLoading
      );

    //  Transfer Lists Actions
    //  Opens transfer menu preference ui dropdown
    case ACTIONS.SET_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE:
      return state.set('showTransferLists', action.showTransferLists);
    //  Toggles one single transfer list item in dropdown
    case ACTIONS.TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE_ON_STATE:
      return state.update('selectedTransferLists', (transferLists) => {
        if (transferLists.includes(action.transferList)) {
          return transferLists.filter(
            (transferList) => transferList !== action.transferList
          );
        } else {
          return transferLists.push(action.transferList);
        }
      });
    //  Toggles all transfer lists items in dropdown
    case ACTIONS.TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE_ON_STATE:
      return state.update('selectedTransferLists', (transferLists) => {
        if (
          action.transferLists.length > 0 &&
          transferLists.size < action.transferLists.length
        ) {
          const newTransferLists = action.transferLists.filter(
            (transferList) => !transferLists.includes(transferList)
          );
          return transferLists.push(...newTransferLists);
        } else {
          return List();
        }
      });
    //  Toggles all local storage transfer lists items
    case ACTIONS.TOGGLE_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE:
      return state.set('selectedTransferLists', List(action.transferLists));
    //  Toggles the loading spinner for transfer lists in collapsibleMultiselect component
    case ACTIONS.SET_PREFERENCE_MENU_TRANSFER_LISTS_LOADING:
      return state.set(
        'preferenceMenuTransferListsLoading',
        action.preferenceMenuTransferListsLoading
      );
    default:
      return state;
  }
}
