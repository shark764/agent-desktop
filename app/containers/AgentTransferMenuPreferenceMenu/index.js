import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CollapsibleMultiselect from 'components/CollapsibleMultiselect';

import { selectQueues } from 'containers/AgentDesktop/selectors';
import { selectUserAssignedTransferLists } from 'containers/TransferMenu/selectors';
import ErrorBoundary from 'components/ErrorBoundary';
import {
  selectAgentsPreferences,
  selectVisibleQueues,
  selectVisibleTransferLists,
  selectPreferenceMenuQueuesLoading,
  selectPreferenceMenuTransferListsLoading,
} from './selectors';
import {
  toggleAgentsTransferMenuPreference,
  toggleSelectedQueueTransferMenuPreference,
  toggleAllSelectedQueuesTransferMenuPreference,
  toggleSelectedTransferListTransferMenuPreference,
  toggleAllSelectedTransferListsTransferMenuPreference,
  toggleShowQueuesTransferMenuPreference,
  toggleShowTransferListsTransferMenuPreference,
  initializeTransferMenuPreferences,
} from './actions';

import messages from './messages';

export class AgentTransferMenuPreferenceMenu extends React.Component {
  constructor(props) {
    super(props);
    this.initializeTransferLists();
  }

  initializeTransferLists = () => {
    this.props.initializeTransferMenuPreferences();
  };

  render() {
    return (
      <Fragment>
        <CollapsibleMultiselect
          title={messages.agents}
          toggleSelection={this.props.toggleAgents}
          singleToggleBtn={this.props.agentsTransferMenuEnabled}
        />

        <CollapsibleMultiselect
          title={messages.queues}
          items={this.props.queues}
          selectAll={this.props.toggleAllQueues}
          selectAllBtn
          selectedItems={this.props.visibleQueues.map((queue) => queue.id)}
          toggleSelection={this.props.toggleQueue}
          toggleShowList={this.props.toggleShowQueues}
          loading={this.props.preferenceMenuQueuesLoading}
        />

        <CollapsibleMultiselect
          title={messages.transferLists}
          items={
            this.props.transferLists !== null ? this.props.transferLists : []
          }
          selectAll={this.props.toggleAllTransferLists}
          selectAllBtn
          selectedItems={this.props.visibleTransferLists.map(
            (transferList) => transferList.id
          )}
          toggleSelection={this.props.toggleTransferList}
          toggleShowList={this.props.toggleShowTransferLists}
          loading={this.props.preferenceMenuTransferListsLoading}
        />
      </Fragment>
    );
  }
}

const mapStateToProps = (state, props) => ({
  agentsTransferMenuEnabled: selectAgentsPreferences(state, props),
  transferLists: selectUserAssignedTransferLists(state, props),
  queues: selectQueues(state, props),
  visibleQueues: selectVisibleQueues(state, props),
  visibleTransferLists: selectVisibleTransferLists(state, props),
  preferenceMenuQueuesLoading: selectPreferenceMenuQueuesLoading(state, props),
  preferenceMenuTransferListsLoading: selectPreferenceMenuTransferListsLoading(
    state,
    props
  ),
});

AgentTransferMenuPreferenceMenu.propTypes = {
  agentsTransferMenuEnabled: PropTypes.bool,
  toggleAgents: PropTypes.func.isRequired,
  transferLists: PropTypes.array,
  queues: PropTypes.array,
  toggleQueue: PropTypes.func,
  toggleAllQueues: PropTypes.func,
  toggleTransferList: PropTypes.func,
  toggleAllTransferLists: PropTypes.func,
  toggleShowQueues: PropTypes.func,
  toggleShowTransferLists: PropTypes.func,
  initializeTransferMenuPreferences: PropTypes.func.isRequired,
  visibleQueues: PropTypes.array,
  visibleTransferLists: PropTypes.array,
  preferenceMenuQueuesLoading: PropTypes.bool,
  preferenceMenuTransferListsLoading: PropTypes.bool,
};

export const actions = {
  toggleAgents: toggleAgentsTransferMenuPreference,
  toggleQueue: toggleSelectedQueueTransferMenuPreference,
  toggleAllQueues: toggleAllSelectedQueuesTransferMenuPreference,
  toggleTransferList: toggleSelectedTransferListTransferMenuPreference,
  toggleAllTransferLists: toggleAllSelectedTransferListsTransferMenuPreference,
  toggleShowQueues: toggleShowQueuesTransferMenuPreference,
  toggleShowTransferLists: toggleShowTransferListsTransferMenuPreference,
  initializeTransferMenuPreferences,
};

export default ErrorBoundary(
  connect(
    mapStateToProps,
    actions
  )(AgentTransferMenuPreferenceMenu)
);
