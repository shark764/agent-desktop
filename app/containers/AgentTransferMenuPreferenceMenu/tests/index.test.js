/*
 * Copyright Â© 2015-2019 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { AgentTransferMenuPreferenceMenu } from 'containers/AgentTransferMenuPreferenceMenu';

describe('<AgentTransferMenuPreferenceMenu />', () => {
  describe('agent preference menu enabled', () => {
    const rendered = shallow(
      <AgentTransferMenuPreferenceMenu
        initializeTransferMenuPreferences={() => {}}
        agentsTransferMenuEnabled
        toggleAgents={() => {}}
        toggleQueue={() => {}}
        toggleTransferList={() => {}}
        visibleQueues={[]}
        visibleTransferLists={[]}
      />
    );
    it('renders the multiple preference options with select all button and open by default', () => {
      expect(rendered).toMatchSnapshot();
    });
  });
  describe('renders queues multi item preference list with selected items', () => {
    describe('preference list rendered', () => {
      const rendered = shallow(
        <AgentTransferMenuPreferenceMenu
          initializeTransferMenuPreferences={() => {}}
          queues={['123', 'abc']}
          toggleQueue={() => {}}
          toggleAllQueues={() => {}}
          toggleShowQueues={() => {}}
          visibleQueues={['123', 'abc']}
          visibleTransferLists={[]}
          preferenceMenuQueuesLoading={false}
          toggleAgents={() => {}}
          toggleTransferList={() => {}}
          toggleAllTransferLists={() => {}}
          toggleShowTransferLists={() => {}}
        />
      );
      it('renders the multiple preference options with select all button', () => {
        expect(rendered).toMatchSnapshot();
      });
    });
  });
  describe('renders queues multi item preference list with loading state and empty list and loading state', () => {
    describe('preference list not rendered', () => {
      const rendered = shallow(
        <AgentTransferMenuPreferenceMenu
          initializeTransferMenuPreferences={() => {}}
          queues={[]}
          toggleQueue={() => {}}
          toggleAllQueues={() => {}}
          toggleShowQueues={() => {}}
          visibleQueues={[]}
          visibleTransferLists={[]}
          preferenceMenuQueuesLoading
          toggleAgents={() => {}}
          toggleTransferList={() => {}}
          toggleAllTransferLists={() => {}}
          toggleShowTransferLists={() => {}}
        />
      );
      it('does not renders the multiple preference options and sets loading state for queues list', () => {
        expect(rendered).toMatchSnapshot();
      });
    });
  });
  describe('renders transfer lists multi item preference list with selected items', () => {
    describe('preference list rendered', () => {
      const rendered = shallow(
        <AgentTransferMenuPreferenceMenu
          initializeTransferMenuPreferences={() => {}}
          transferLists={['123', 'abc']}
          toggleTransferList={() => {}}
          toggleAllTransferLists={() => {}}
          toggleShowTransferLists={() => {}}
          visibleTransferLists={['123', 'abc']}
          visibleQueues={[]}
          preferenceMenuTransferListsLoading={false}
          toggleAgents={() => {}}
          toggleQueue={() => {}}
          toggleAllQueues={() => {}}
          toggleShowQueues={() => {}}
        />
      );
      it('renders the multiple preference options with select all button', () => {
        expect(rendered).toMatchSnapshot();
      });
    });
  });
  describe('renders transfer lists multi item preference list with loading state and empty list', () => {
    describe('preference list not rendered', () => {
      const rendered = shallow(
        <AgentTransferMenuPreferenceMenu
          initializeTransferMenuPreferences={() => {}}
          transferLists={[]}
          toggleTransferList={() => {}}
          toggleAllTransferLists={() => {}}
          toggleShowTransferLists={() => {}}
          visibleTransferLists={[]}
          visibleQueues={[]}
          preferenceMenuTransferListsLoading
          toggleAgents={() => {}}
          toggleQueue={() => {}}
          toggleAllQueues={() => {}}
          toggleShowQueues={() => {}}
        />
      );
      it('does not renders the multiple preference options and sets loading state for transfer lists items', () => {
        expect(rendered).toMatchSnapshot();
      });
    });
  });
});
