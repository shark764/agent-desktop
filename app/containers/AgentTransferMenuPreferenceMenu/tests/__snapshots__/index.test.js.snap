// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgentTransferMenuPreferenceMenu /> agent preference menu enabled renders the multiple preference options with select all button and open by default 1`] = `
<React.Fragment>
  <CollapsibleMultiselect
    easing="linear"
    singleToggleBtn={true}
    title={
      Object {
        "defaultMessage": "Agents",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.agents",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Queues",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.queues",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Transfer Lists",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.transferLists",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
</React.Fragment>
`;

exports[`<AgentTransferMenuPreferenceMenu /> renders queues multi item preference list with loading state and empty list and loading state preference list not rendered does not renders the multiple preference options and sets loading state for queues list 1`] = `
<React.Fragment>
  <CollapsibleMultiselect
    easing="linear"
    title={
      Object {
        "defaultMessage": "Agents",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.agents",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    items={Array []}
    loading={true}
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Queues",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.queues",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Transfer Lists",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.transferLists",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
</React.Fragment>
`;

exports[`<AgentTransferMenuPreferenceMenu /> renders queues multi item preference list with selected items preference list rendered renders the multiple preference options with select all button 1`] = `
<React.Fragment>
  <CollapsibleMultiselect
    easing="linear"
    title={
      Object {
        "defaultMessage": "Agents",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.agents",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    items={
      Array [
        "123",
        "abc",
      ]
    }
    loading={false}
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={
      Array [
        undefined,
        undefined,
      ]
    }
    title={
      Object {
        "defaultMessage": "Queues",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.queues",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Transfer Lists",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.transferLists",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
</React.Fragment>
`;

exports[`<AgentTransferMenuPreferenceMenu /> renders transfer lists multi item preference list with loading state and empty list preference list not rendered does not renders the multiple preference options and sets loading state for transfer lists items 1`] = `
<React.Fragment>
  <CollapsibleMultiselect
    easing="linear"
    title={
      Object {
        "defaultMessage": "Agents",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.agents",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Queues",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.queues",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    items={Array []}
    loading={true}
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Transfer Lists",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.transferLists",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
</React.Fragment>
`;

exports[`<AgentTransferMenuPreferenceMenu /> renders transfer lists multi item preference list with selected items preference list rendered renders the multiple preference options with select all button 1`] = `
<React.Fragment>
  <CollapsibleMultiselect
    easing="linear"
    title={
      Object {
        "defaultMessage": "Agents",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.agents",
      }
    }
    toggleSelection={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={Array []}
    title={
      Object {
        "defaultMessage": "Queues",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.queues",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
  <CollapsibleMultiselect
    easing="linear"
    items={
      Array [
        "123",
        "abc",
      ]
    }
    loading={false}
    selectAll={[Function]}
    selectAllBtn={true}
    selectedItems={
      Array [
        undefined,
        undefined,
      ]
    }
    title={
      Object {
        "defaultMessage": "Transfer Lists",
        "id": "app.containers.AgentTransferMenuPreferenceMenu.transferLists",
      }
    }
    toggleSelection={[Function]}
    toggleShowList={[Function]}
    transitionTime={400}
  />
</React.Fragment>
`;
